File: single-language-analysis/dataset-files/java/TestSmell139.java
import org.junit.Test;
import static org.junit.Assert.*;

public class TestSmell139 {

    @Test
    public void testPluginLoadingAndUsage() {
        ThreadSafeScheduler scheduler = new ThreadSafeScheduler();
        List<String> results = new ArrayList<>();
        
        scheduler.loadPlugin("p1", results::add);
        
        scheduler.schedule("jobp", () -> scheduler.getPlugins().get("p1").accept("ok"), 1);
        scheduler.scheduleOneOff("onep", () -> scheduler.getPlugins().get("p1").accept("ok"), 
            LocalDateTime.now().plusSeconds(1));
        
        try {
            Thread.sleep(1100);
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }
        
        scheduler.runPending();
        
        try {
            Thread.sleep(100);
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }
        
        assertEquals(2, Collections.frequency(results, "ok"));
    }
}

File: single-language-analysis/dataset-files/python/test_smell_139.py
# Original URL: https://github.com/code-refactor/schedualpy_dataeng_alex/blob/493c004678c9dcf408a12e7faf17daa35dd7be81/test_plugins.py#L4-L19

def test_smell_139():
    scheduler = ThreadSafeScheduler()
    results = []
    def plugin(data):
        results.append(data)
    scheduler.load_plugin('p1', plugin)
    # simulate using plugin in post-hook
    def job(): scheduler.plugins['p1']('ok')
    scheduler.schedule('jobp', job, cron_expr=1)
    scheduler.schedule_one_off('onep', job, run_at=__import__('datetime').datetime.now() + __import__('datetime').timedelta(seconds=1))
    time = __import__('time')
    time.sleep(1.1)
    scheduler._run_pending()
    time.sleep(0.1)
    # two runs: one recurring, one one-off
    assert results.count('ok') == 2